Minor fixes

brbsix opened the pull request.

brbsix committed:
Handle environments in which USER is not set (e.g. Termux)

brbsix committed:
> is for string comparisons. Use -gt instead.

brbsix committed:
Fix error message "ls: cannot access '*': No such file or directory"

To reproduce:

    # install bpm
    rm -rf ~/.bpm
    git clone https://github.com/netj/bpm.git ~/.bpm

    shopt -u nullglob
    . ~/.bpm/bpm.sh

This is resolved by enabling nullglob.

I am suppressing stderr from `ls` as well.

brbsix committed:
Prevent word splitting

brbsix committed:
I believe this achieves the desired effect

netj commented on bpm.sh:
```diff
@@ -35,7 +35,7 @@ if ${BPM_LOADED:-false}; then
     __bpm_info() { :; }
 fi
-shopt -s extglob
+shopt -s extglob nullglob
```
```suggestion
shopt -s extglob
```
I'm not entirely comfortable with nullglob for all my sessions.  Is this mainly for the `ls -tdL ... ./*` below?  Then I'd suggest we turn it on for that command only.

netj commented on bpm.sh:
```diff
@@ -165,10 +165,10 @@ __bpm_list_enabled_by_deps() {
     # echo $latest >&2
     deps="$BPM_TMPDIR"/enabled.deps
-    if [[ "$deps" -nt $latest ]]; then
+    if [[ -h $latest && "$deps" -nt $latest ]]; then
```
Not sure I follow this part.  Let me take another stab.  I think this was to detect any `bpm on/off` as well as enabled plugins getting updated since enabled.deps was last generated.

Hope you don't mind my pushing commits to this PR :)

netj committed:
Smoother enabled.deps refresh handling

Revert "I believe this achieves the desired effect"

This reverts commit 0207dba136e95cac0a232a64f59e2dc38b4bb001.

Revert "Fix error message "ls: cannot access '*': No such file or directory""

This reverts commit c73d2ba2a0e26f5bde7e8ec258aefe80b1b64206.

netj reviewed (approved):
Thanks a lot for these fixes! (and sorry, not sure how I missed your PR for months...)

Let me add a small commit addressing my concerns and merge this.

netj committed:
quotes
